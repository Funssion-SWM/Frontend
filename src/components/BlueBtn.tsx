type Props = {
  text: string;
  onClick: () => void;
  extraStyle?: string;
};

export default function BlueBtn({ text, onClick, extraStyle = '' }) {
  return (
    <button
      className={`bg-blue-500 text-white px-3 py-1 rounded-2xl ${extraStyle}`}
      onClick={() => onClick()}
    >
      {text}
    </button>
  );
}
