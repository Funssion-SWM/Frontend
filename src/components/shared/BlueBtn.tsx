type Props = {
  text: string;
  onClick: () => void;
  extraStyle?: string;
};

export default function BlueBtn({ text, onClick, extraStyle = '' }) {
  return (
    <button
      className={`bg-soma-blue-40 text-white px-3 py-1 rounded-2xl ${extraStyle}`}
      onClick={() => onClick()}
    >
      {text}
    </button>
  );
}
