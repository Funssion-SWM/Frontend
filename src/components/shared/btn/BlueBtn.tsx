import { ButtonSize } from '@/types';

type Props = {
  text: string;
  onClick: () => void;
  size?: ButtonSize;
  extraStyle?: string;
};

export default function BlueBtn({
  text,
  onClick,
  size = 'medium',
  extraStyle = '',
}: Props) {
  return (
    <button
      className={`bg-soma-blue-40 text-white px-3.5 py-2 rounded-3xl transition hover:bg-soma-blue-50 ${
        {
          big: 'text-sm sm:text-base',
          medium: 'text-xs sm:text-sm',
          small: 'text-[10px] sm:text-xs',
        }[size]
      }  ${extraStyle} `}
      onClick={() => onClick()}
    >
      {text}
    </button>
  );
}
