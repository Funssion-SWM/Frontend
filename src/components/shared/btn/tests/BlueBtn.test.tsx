import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import BlueBtn from '../BlueBtn';
import userEvent from '@testing-library/user-event';

describe('Blue Btn', () => {
  it('renders correctly when size is small', () => {
    const component = renderer.create(
      <BlueBtn text="text" size="small" onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when size is medium', () => {
    const component = renderer.create(
      <BlueBtn text="text" size="medium" onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when size is big', () => {
    const component = renderer.create(
      <BlueBtn text="text" size="big" onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('when button clicked', async () => {
    const mockCallBack = jest.fn();

    render(<BlueBtn text="text" onClick={mockCallBack} />);

    const btn = screen.getByRole('button');
    await userEvent.click(btn);
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
