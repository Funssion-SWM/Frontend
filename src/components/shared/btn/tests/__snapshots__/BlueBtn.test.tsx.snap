// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blue Btn renders correctly 1`] = `
<button
  className="bg-soma-blue-40 text-white px-3.5 py-2 rounded-3xl transition hover:bg-soma-blue-50 text-sm sm:text-base  "
  onClick={[Function]}
>
  text
</button>
`;
